From 9bbff7e6b0ce00e4df925048eb74f8c02f998019 Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sebastian.reichel@collabora.com>
Date: Thu, 6 Aug 2026 18:36:25 +0200
Subject: [PATCH 131/293] usb: typec: fusb302: drop explicit DRM DP HPD bridge
 support

There is now generic USB Type-C DP HPD bridge support via
CONFIG_DRM_AUX_HPD_TYPEC_BRIDGE, which is automatically enabled when
CONFIG_DRM_AUX_HPD_BRIDGE and TYPEC are enabled. The manual explicit
registration code has the same requirements and thus is pointless and
can be dropped.

Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
---
 drivers/usb/typec/tcpm/Kconfig   |  2 --
 drivers/usb/typec/tcpm/fusb302.c | 11 -----------
 2 files changed, 13 deletions(-)

diff --git a/drivers/usb/typec/tcpm/Kconfig b/drivers/usb/typec/tcpm/Kconfig
index 00baa7503..8cdd84ca5 100644
--- a/drivers/usb/typec/tcpm/Kconfig
+++ b/drivers/usb/typec/tcpm/Kconfig
@@ -58,8 +58,6 @@ config TYPEC_FUSB302
 	tristate "Fairchild FUSB302 Type-C chip driver"
 	depends on I2C
 	depends on EXTCON || !EXTCON
-	depends on DRM || DRM=n
-	select DRM_AUX_HPD_BRIDGE if DRM_BRIDGE && OF
 	help
 	  The Fairchild FUSB302 Type-C chip driver that works with
 	  Type-C Port Controller Manager to provide USB PD and USB
diff --git a/drivers/usb/typec/tcpm/fusb302.c b/drivers/usb/typec/tcpm/fusb302.c
index f5c0cc453..4566b7daa 100644
--- a/drivers/usb/typec/tcpm/fusb302.c
+++ b/drivers/usb/typec/tcpm/fusb302.c
@@ -5,7 +5,6 @@
  * Fairchild FUSB302 Type-C Chip Driver
  */
 
-#include <drm/bridge/aux-bridge.h>
 #include <linux/debugfs.h>
 #include <linux/delay.h>
 #include <linux/devm-helpers.h>
@@ -1685,7 +1684,6 @@ static int fusb302_probe(struct i2c_client *client)
 {
 	struct fusb302_chip *chip;
 	struct i2c_adapter *adapter = client->adapter;
-	struct auxiliary_device *bridge_dev;
 	struct device *dev = &client->dev;
 	const char *name;
 	int ret = 0;
@@ -1749,11 +1747,6 @@ static int fusb302_probe(struct i2c_client *client)
 	if (ret < 0)
 		return ret;
 
-	bridge_dev = devm_drm_dp_hpd_bridge_alloc(chip->dev, to_of_node(chip->tcpc_dev.fwnode));
-	if (IS_ERR(bridge_dev))
-		return dev_err_probe(chip->dev, PTR_ERR(bridge_dev),
-				     "failed to alloc bridge\n");
-
 	chip->tcpm_port = devm_tcpm_register_port(&client->dev, &chip->tcpc_dev);
 	if (IS_ERR(chip->tcpm_port))
 		return dev_err_probe(dev, PTR_ERR(chip->tcpm_port),
@@ -1777,10 +1770,6 @@ static int fusb302_probe(struct i2c_client *client)
 
 	i2c_set_clientdata(client, chip);
 
-	ret = devm_drm_dp_hpd_bridge_add(chip->dev, bridge_dev);
-	if (ret)
-		return ret;
-
 	enable_irq_wake(chip->irq);
 
 	return ret;
-- 
2.53.0

